src/core/net/netif.cpp

```diff
@@ -209,37 +218,36 @@ ThreadError Netif::UnsubscribeMulticast(const NetifMulticastAddress &aAddress)
 ThreadError Netif::SubscribeExternalMulticast(const Address &aAddress)
 {
     ThreadError error = kThreadError_None;
-    int8_t index = 0;
+    NetifMulticastAddress *entry;
+    size_t num = sizeof(mExtMulticastAddresses) / sizeof(mExtMulticastAddresses[0]);
```

Comment:
Not really... Just a style preference... I think this may make it just a bit more flexible (e.g., during testing we may manually hardcore/override the config value in the src...)